### PR TITLE
add language reference table + nits

### PR DESCRIPTION
**Pull request by cartersocha**
Based on previous PR discussion. I'm a bit worried about not including signal info as well but we can mature as we go.

### CODE REVIEW

**Comment by cartersocha** [comment]
Referencing this PR: https://github.com/open-telemetry/opentelemetry-demo/pull/431

**Comment by cartersocha** [comment]
@mviitane I think I added the table like you suggested. @austinlparker as an FYI

File: docs/README.md [hunk 1]
```diff
@@ -13,6 +13,24 @@ Want to deploy the demo and see it in action? Start here.
...
 - [Kubernetes](./kubernetes_deployment.md)
 
+### Language Feature Reference
+
+Want to understand how a particular language's instrumentation works? Start
+here.
+
+| Language      | Auto Instrumentation                                                                                                | Manual Instrumentation                                                                                              |
+|---------------|---------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------|
+| Cpp           |                                                                                                                     |                                                                                                                     |
```
Comment (mviitane) [inline docs/README.md:23]: I'd use C++ and .NET formats. However, we seem to have some variance in other tables.

**Review by mviitane — APPROVED** [review]

**Comment by mviitane** [comment]
> @mviitane I think I added the table like you suggested. @austinlparker as an FYI

Yes, this is good 👍

File: docs/README.md (hunk 1 above)
Comment (austinlparker) [reply docs/README.md:23]: I also like C++ and .NET

**Review by austinlparker — APPROVED** [review]
lgtm outside of comments

File: docs/README.md (hunk 1 above)
Comment (reyang) [reply docs/README.md:23]: +1 `C++` and `.NET`

File: docs/services/frontend.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# frontend
+# Front End
```
Comment (reyang) [inline docs/services/frontend.md:1]: I'm not sure about this one, here are some opinions https://www.designernews.co/stories/42206-frontend-frontend-or-front-end-developer

**Review by reyang — APPROVED** [review]

File: docs/services/frontend.md (hunk 2 above)
Comment (cartersocha) [reply docs/services/frontend.md]: I'm going to leave it as frontend

**Pull request merged**